R/server_header_summaries.R

```diff
@@ -212,6 +212,82 @@ header_summary.scheme = function(value, ...) { #nolint
                 value = as.character(value))
 }
 
+#' @rdname header_summary
+#' @export
+`header_summary.cookies` = function(value, ...) { #nolint
+  security_header = class(value)
+  value = as.logical(value)
+
+  if (length(value) == 0 || all(is.na(value))) {
+    status = "OK"
+    message = "No cookies detected"
+  } else if (all(value)) {
+    status = "OK"
+    message = "All cookies use the Secure flag"
+  } else {
+    status = "WARN"
+    message = "Cookies set without using the Secure flag"
+  }
+
+  dplyr::tibble(security_header = security_header,
+                status = status,
+                message = message,
+                value = NA_character_)
+}
+
+#' @rdname header_summary
+#' @export
+`header_summary.redirection` = function(value, ...) { #nolint
+  security_header = class(value)
+  value = as.logical(value)
+  if (value) {
+    status = "OK"
+    message = "Redirects to HTTPS"
+  } else {
+    status = "WARN"
+    message = "Does not redirect to an HTTPS site"
+  }
+
+  dplyr::tibble(security_header = security_header,
+                status = status,
+                message = message,
+                value = NA_character_)
+}
+
+#' @rdname header_summary
+#' @export
+`header_summary.subresource-integrity` = function(value, ...) { #nolint
+  security_header = class(value)
+
+  if (length(value) == 0) {
+    status = "OK"
+    message = "Subresource Integrity is not needed since site contains no script tags"
+  } else {
+    integrity = purrr::map(value, "integrity") |>
```

Comment:
Would using purrr::map_chr be simplier?

Reply:
Nope, purrr::map_chr() would error if it's NULL (which it is for jumpingrivers.com for example)